src/app/filter/filter.service.spec.ts

```diff
@@ -13,4 +13,37 @@ describe('FilterService', () => {
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
+
+  it('should have values$ Observable', () => {
+    expect(service.values$).toBeDefined();
+  });
+
+  it('should have the right values', () => {
+    service.values$.subscribe(values => {
+      expect(values).toEqual({
+        currentPage: 1,
+        entriesPerPage: 20
+      });
+    });
```

Comment:
if you subscribe in a test case there is the possibility that the test case end before that the expectation have been executed for real.

You have to use one strategy to "wait" that the subscribe happens, I use usually the done() strategy but recently  I discovered that it is not really the best one, you could use fakeAsync()

https://piotrl.medium.com/angular-testing-avoid-done-function-c6ad11df0bf9

TL;DR; the **done()** is really immediate to implement and readable but it is slower than fakeAsync, moreover, fakeAsync allows you to "test" multiple emitted values, done only one (only one subscribe).
I strongly recommend you to use only fakeAsync or a bit of both if you would like to practice and to read the article.